phpBB/viewforum.php

```diff
@@ -47,27 +47,41 @@
 	trigger_error('NO_FORUM');
 }
 
+$sql_ary = array(
+	'SELECT'	=> "f.*",
+	'FROM'		=> array(
+		FORUMS_TABLE		=> 'f'
+	),
+	'WHERE'		=> 'f.forum_id = ' . $forum_id,
+	'LEFT_JOIN'	=> array(),
+);
+
 $sql_from = FORUMS_TABLE . ' f';
 $lastread_select = '';
 
 // Grab appropriate forum data
 if ($config['load_db_lastread'] && $user->data['is_registered'])
 {
-	$sql_from .= ' LEFT JOIN ' . FORUMS_TRACK_TABLE . ' ft ON (ft.user_id = ' . $user->data['user_id'] . '
-		AND ft.forum_id = f.forum_id)';
-	$lastread_select .= ', ft.mark_time';
+	$sql_ary['LEFT_JOIN'][] = array('FROM' => array(FORUMS_TRACK_TABLE => 'ft'), 'ON' => 'ft.user_id = ' . $user->data['user_id'] . ' AND ft.forum_id = f.forum_id');
+	$sql_ary['SELECT'] .= ', ft.mark_time';
 }
 
 if ($user->data['is_registered'])
 {
-	$sql_from .= ' LEFT JOIN ' . FORUMS_WATCH_TABLE . ' fw ON (fw.forum_id = f.forum_id AND fw.user_id = ' . $user->data['user_id'] . ')';
-	$lastread_select .= ', fw.notify_status';
+	$sql_ary['LEFT_JOIN'][] = array('FROM' => array(FORUMS_WATCH_TABLE => 'fw'), 'ON' => 'fw.forum_id = f.forum_id AND fw.user_id = ' . $user->data['user_id']);
+	$sql_ary['SELECT'] .= ', fw.notify_status';
 }
 
-$sql = "SELECT f.* $lastread_select
-	FROM $sql_from
-	WHERE f.forum_id = $forum_id";
-$result = $db->sql_query($sql);
+/**
+* You can use this event to modify the sql used to select the forum on the viewforum page.
+*
+* @event core.viewforum_modify_sql
+* @var array	sql_ary		The SQL array to get the data of all topics
+* @since 3.2.2-RC1
```

Comment:
This is incorrect. 3.2 is EOL so you need to pick a beta release for 3.3.

Reply:
I updated this to 3.3.14 but I'm not sure if that's the right version to use; let me know if I should pick a different version number here!